src/bcccp/tickets/season/SeasonTicket.java

```diff
@@ -18,61 +19,98 @@ public SeasonTicket (String ticketId,
 			             String carparkId, 
 			             long startValidPeriod,
 			             long endValidPeriod) {
-		//TDO Implement constructor
+
+		this.ticketId = ticketId;
+		this.carparkId = carparkId;
+		this.startValidPeriod = startValidPeriod;
+		this.endValidPeriod = endValidPeriod;
+
+		usages = new ArrayList<>();
 	}
 
 	@Override
 	public String getId() {
-		// TODO Auto-generated method stub
-		return null;
+		return ticketId;
 	}
 
 	@Override
 	public String getCarparkId() {
-		// TODO Auto-generated method stub
-		return null;
+		return carparkId;
 	}
 
 	@Override
 	public long getStartValidPeriod() {
-		// TODO Auto-generated method stub
-		return 0;
+		return startValidPeriod;
 	}
 
 	@Override
 	public long getEndValidPeriod() {
-		// TODO Auto-generated method stub
-		return 0;
+		return endValidPeriod;
 	}
 
 	@Override
 	public boolean inUse() {
-		// TODO Auto-generated method stub
-		return false;
+
+		Iterator<IUsageRecord> usageRec = usages.iterator();
+
+		boolean foundUsageRecord = false;
+
+		while (usageRec.hasNext()) {
+
+		    if (usageRec.next().getSeasonTicketId().equals(getId())) {
+
+                foundUsageRecord = true;
+            }
+
+            else foundUsageRecord = false;
```

Comment:
same to my 'suboptimal' code in adhocticketdao. it's not something you necessarily remember to insert the break after 'hit' in a while loop when writing up the 1st draft. i would rework your method to:

@Override
  	public boolean inUse() {
	
 		boolean foundUsageRecord = false;
 
 		Iterator<IUsageRecord> usageRec = usages.iterator();
 
 		while (usageRec.hasNext()) {
 
 		    if (usageRec.next().getSeasonTicketId().equals(getId())) {
 
                 foundUsageRecord = true;

                 break;
           }